src/goto-instrument/contracts/utils.cpp

```diff
@@ -142,12 +142,13 @@ const symbolt &new_tmp_symbol(
   const typet &type,
   const source_locationt &location,
   const irep_idt &mode,
-  symbol_table_baset &symtab)
+  symbol_table_baset &symtab,
+  std::string suffix)
 {
   return get_fresh_aux_symbol(
     type,
-    id2string(location.get_function()) + "::tmp_cc",
-    "tmp_cc",
+    id2string(location.get_function()) + "::",
+    suffix,
     location,
     mode,
     symtab);
```

Comment:
While at it: should this code also be modified to have those symbols show up in counterexample traces? I found myself having a hard time understanding counterexamples for none of the assignments to these variables show up, but failing assertions then refer to them. The change required to facilitate this:
````c++
symbolt &new_symbol = get_fresh_aux_symbol(...);
new_symbol.is_auxiliary = false;
return new_symbol;
````

Reply:
Adding a new function argument 'is_auxiliary' to control that symbol per symbol.